src/AdminCli/Configuration/AppSettings.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Serilog.Events;
+
+namespace AdminCli.Configuration;
+
+public sealed class AppSettings
+{
+    public const string DefaultLocalGatewayUrl = "http://locahost:5000";
+    public const string DefaultCloudGatewayUrl = "https://cn.thinktecture-demos.com/api";
+    public IdentityServerSettings IdentityServerSettings { get; set; } = new ();
+    public string GatewayUrl { get; set; } = DefaultLocalGatewayUrl;
+    public TokenInfo? CurrentTokenInfo { get; set; }
+    public LogEventLevel LogLevel { get; set; } = LogEventLevel.Warning;
+
+    public bool TryGetAccessToken([NotNullWhen(true)] out string? accessToken)
+    {
+        var now = DateTime.UtcNow;
+        if (CurrentTokenInfo is null ||
+            CurrentTokenInfo.ValidTo.AddSeconds(-30) < now)
```

Comment:
I would push that `30` to an app setting. In this particular case, 30 seconds is a lot... and my guess is that `10` or even `5` should be more than enough. `TryGetAccessToken` is invoked as part of an actual command invocation, so `10` should be okay